launchable/commands/subset.py

```diff
@@ -263,18 +272,36 @@ def subset(
     session_id = None
     tracking_client = TrackingClient(Tracking.Command.SUBSET, app=app)
     try:
-        session_id = find_or_create_session(
-            context=context,
-            session=session,
-            build_name=build_name,
-            flavor=flavor,
-            is_observation=is_observation,
-            links=links,
-            is_no_build=is_no_build,
-            lineage=lineage,
-            tracking_client=tracking_client,
-            test_suite=test_suite,
+        if session_name:
+            if not build_name:
+                raise click.UsageError(
+                    '--build option is required when you use a --session-name option ')
+            sub_path = "builds/{}/test_session_names/{}".format(build_name, session_name)
+            client = LaunchableClient(test_runner=context.invoked_subcommand, app=context.obj, tracking_client=tracking_client)
+            res = client.request("get", sub_path)
+            res.raise_for_status()
+            session_id = "builds/{}/test_sessions/{}".format(build_name, res.json().get("id"))
```

Comment:
Perhaps, it would be better to check if "id" exists?
```suggestion
            id = res.json().get("id", None)
            if id:
                session_id = "builds/{}/test_sessions/{}".format(build_name, res.json().get("id"))
```

Reply:
I don't think so. Because if the test session doesn't exist, the backend returns 4xx.
Then, `raise_for_status()` raise exception. ref: https://github.com/psf/requests/blob/main/src/requests/models.py#L1015